graph/src/schema/sql.graphql

```diff
@@ -0,0 +1,13 @@
+scalar String
+scalar JSONObject
+
+
+# The type names are purposely awkward to minimize the risk of them
+# colliding with user-supplied types
+"The type for the top-level _sql field"
+type _SqlQueryResult_ {
+  "The columns returned by the query"
+  columns: [String!]!
+  "The rows returned by the query"
+  rows: JSONObject!
```

Comment:
What do you think about [[String | Number | ID | BigInt ]!]! as  you have already experienced with the redundant columns as object keys/property names.

Reply:
Right now, JSONObject scalar already exists, so it's easier to use in the mvp.

We can and should have a way to send csv rather than json because it's cheaper. But this can be provided as an option. What do you think about it?

Reply:
If we can send the structure in form of columns and rows as I explained any client can use the result to convert into csv.
I don't think we should implement formatting to result just now.

Reply:
> What do you think about [[String | Number | ID | BigInt ]!]!

this would still be a new union type we create. Doesn't help much to the end user in terms of type safety when parsing

Reply:
I think this is more related to duplication of data, it's more network efficient if we use csv instead of json.

@saihaj do you think we could do anything in our schema to use json today and enable csv in the future without breaking changes?

Reply:
I suppose our return type can be union which will be more flexible and we can even offer better error experience


```graphql
union SqlOutput = SqlJSONOutput | SqlCSVOutput | UnsupportedSqlError

type SqlJSONOutput {
  """
  The SQL query result rows. Each row is represented as a list of values.
  The values are represented as JSON values.
  """
  rows: [Json]
  """
  The SQL query result row count.
  """
  rowCount: Int!
}

type SqlCSVOutput {
  """
  The SQL query result rows. Represented as a CSV string
  """
  result: String
  """
  Number of SQL rows being returned
  """
  rowCount: Int!
}

"""
Return the error if the instance does not support SQL results
"""
type UnsupportedSqlError  implements Error {
  """
  Descriptive message of error
  """
  message: String!
}

```

now as a user querying can do. Now we can even offer querying both JSON or CSV format at same time but we are also offering user a better way to grab errors.
```graphql
query AllUsers {
  sql(input: { query: "SELECT * FROM 'Users'" }) {
    __typename
    ... on SqlJSONOutput {
      rows
      rowCount
    }
    ... on SqlCSVOutput {
      result
    }
    ... on Error {
      message
    }
  }
}
```

Reply:
Created issue #16